### DISCUSSION TITLE
@types/phplint

### DISCUSSION THREAD

**Discussion by Dulux-Oz** [DefinitelyTyped/DefinitelyTyped#58348] [Request a new @types package] [answered]
Hi All,

Could someone help a (TypeScript) noob (again :-) ) with some advice, please, by answering one (or more) of the following questions:

1. Does anyone have the TypeScript def file for `phplint`, and may I please have a copy?
2. Could someone please write the TypeScript def file for `phplint`?
3. Could someone help me to write the TypeScript def file for `phplint`?
4. Could someone point me towards a different PHP linter that works with Gulp and which is TypeScript-useable?
5. Could someone point me towards how to use `phplint` "as it is" with TypeScript?

(FTR: I looked on NPM but couldn't find anything "newer" or "more appropriate" - unlike the mistake I made last time I requested some help ie with `gulp-eslint` and `gulp-eslint-new`.)

Thanks in advance (& thanks again for the previous help I've been given - I really do appreciate it),

Dulux-Oz

**Comment by MysteryBlokHed** [comment, marked as answer]
I just made PR #58379 to add phplint types

**Reply by Dulux-Oz** [reply]
Thank you, I *really* appreciate it.

**Reply by Dulux-Oz** [reply]
Hi,
Just encountered an error:
```
error TS2349: This expression is not callable.
  Type 'typeof import("C:/My_Project/node_modules/@types/phplint/index")' has no call signatures.

  .pipe(phplint())
```
Any ideas?

**Reply by MysteryBlokHed** [reply]
Could you send the code that caused the error?

**Reply by Dulux-Oz** [reply]
That practically is the code that caused the error.  :-)  However, the full task stream (and accompanying imports, etc):
```
import gulp from "gulp";
import phplint from "phplint";

export default function phplinttask(){
	return gulp.src("src/**/*.php")
		.pipe(phplint())
		.pipe(gulp.dest("dest"));
};
```
For the record this is the same pattern I've been using for all of my gulp tasks (html-lint, eslint, ts_transpile, etc, etc, etc) and they all work 100%, which is why I thought this error was caused by the @types def file - that and the error message, of course.

Cheers

Dulux-Oz

**Reply by MysteryBlokHed** [reply]
phplint doesn't have a default export, so `import phplint from "phplint"` ends up doing something similar to `import * as phplint from "phplint"`, which gives an object with the module's exports.
I personally haven't used Grunt/Gulp, but there's a short section about it in phplint's README:

https://github.com/wayneashleyberry/phplint#gulp

**Reply by Dulux-Oz** [reply]
Cool. Thanks.

My bad. All good now.

Sorry for the hassle.  :-)
